packages/cozy-client/src/hooks/useFetchShortcut.jsx

```diff
@@ -0,0 +1,47 @@
+import { useState, useEffect } from 'react'
+
+const useFetchShortcut = (client, id) => {
+  const [shortcutInfos, setShortcutInfos] = useState()
+  const [shortcurtImg, setShotcutImg] = useState()
+  const [fetchStatus, setFetchStatus] = useState('idle')
+  const [shouldDisplayImg, setShouldDisplayImg] = useState(false)
+  useEffect(() => {
+    let timeout
+    const fetchData = async () => {
+      setFetchStatus('loading')
+      try {
+        const shortcutInfosResult = await client
+          .getStackClient()
+          .fetchJSON('GET', `/shortcuts/${id}`)
+        const shortcutRemoteUrl = new URL(
+          shortcutInfosResult.data.attributes.url
+        )
+
+        const imgUrl = `${client.getStackClient().uri}/bitwarden/icons/${
+          shortcutRemoteUrl.host
+        }/icon.png`
+
+        setShotcutImg(imgUrl)
+        // this is used in conjonction with CSS display none/block to
+        // make the http request to load the image and not having a blank
+        // element for a few ms
+        timeout = setTimeout(() => setShouldDisplayImg(true), 400)
```

Comment:
I don't understand this, why wait 400ms after getting the image to display it?

Reply:
You have the choice to display it before or not. I use it like a kind of preloader. Let's take a look at the current flow : 

- App => makes a request to /shortcut => wait a few ms to get the result from the stack => setShortCutInfo and build url => component refresh and replace the placeholder by <img src=shortcutinfo.url>. Then the app makes a request to shortcutinfo.url to get the image and then display it. 

Between those two last action, we display a blank space... The idea of this timeout is to reduce this blank and preload the image.

Reply:
As discussed together, we will remove it from the hook itself but create a new component to preload an image.